src/tree-selector/treeSelectorComponent.ts

```diff
@@ -0,0 +1,52 @@
+import * as ng from 'angular';
+export class TreeSelectorController {
+
+  public data: any;
+  public selectables: any;
+  public display: any;
+  public onSelect: any;
+  public lazyLoad: any;
+
+  /*@ngInject*/
+  constructor(private $http, private $timeout : ng.ITimeoutService) {}
+
+  public $onInit() {
+    this.data = this.parseSelectable(this.data);
+  }
+
+  public handleLazyLoad = (node) => {
+    const dataPromise = this.lazyLoad(node.key);
+    return dataPromise.then((data) => {
+      return this.parseSelectable(data);
+    });
+  }
+
+  public parseSelectable = (data: any) => {
+    return data.map((node, key) => {
+      const parsedData = {...node};
+      if(parsedData.nodes) {
+        parsedData.nodes = this.parseSelectable(parsedData.nodes);
+      }
+      parsedData.selectable = !!parsedData.key.match(this.selectables);
+      return parsedData;
+    });
+  }
+
+  public handleSelect = (data) => {
+    this.$timeout(() => this.onSelect(data));
+    this.display(false);
+  }
+}
+
+export default class TreeSelector implements ng.IComponentOptions {
+  public controller = TreeSelectorController;
+  public controllerAs = 'vm';
+  public template = require('./treeSelector.html');
+  public bindings: any = {
+    data: '<',
```

Comment:
I'm not sure about the `<` here as the `TreeView` component has a `@`, but maybe I'm missing something.